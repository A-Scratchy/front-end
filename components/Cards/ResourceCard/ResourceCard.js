import React from 'react';
import { string, number, func, oneOf } from 'prop-types';
import classnames from 'classnames';

import Card from 'components/Cards/Card/Card';
import ThumbsUp from 'static/images/icons/FontAwesome/thumbs-up.svg';
import ThumbsDown from 'static/images/icons/FontAwesome/thumbs-down.svg';
import styles from './ResourceCard.css';

ResourceCard.propTypes = {
  name: string.isRequired,
  imageSource: string.isRequired,
  description: string,
  userVote: oneOf(['upvote', 'downvote', '']),
  upvotes: number,
  onUpvote: func,
  downvotes: number,
  onDownvote: func,
  href: string,
  className: string,
};

ResourceCard.defaultProps = {
  className: undefined,
  description: '',
  userVote: '',
  upvotes: 0,
  onUpvote: () => {},
  downvotes: 0,
  onDownvote: () => {},
  href: '',
};

export default function ResourceCard({
  imageSource,
  name,
  description,
  userVote,
  upvotes,
  onUpvote,
  downvotes,
  onDownvote,
  className,
}) {
  return (
    <Card className={classnames(styles.ResourceCard, className)}>
      <header className={styles.titleSection}>
        <img src={imageSource} alt="logo" />
        <h5>{name}</h5>
      </header>

      <section>
        <p className={styles.descriptionText}>{description}</p>
      </section>

      <footer className={styles.footerSection}>
        <span className={styles.footerText}>I found this useful </span>
        <button
          type="button"
          className={classnames(styles.voteBtn, { [styles.faded]: !upvotes })}
          onClick={onUpvote}
        >
          <ThumbsUp
            className={classnames(styles.icon, { [styles.active]: userVote === 'upvote' })}
          />
          {upvotes}
        </button>

        <button
          type="button"
          className={classnames(styles.voteBtn, { [styles.faded]: !downvotes })}
          onClick={onDownvote}
        >
          <ThumbsDown
            className={classnames(styles.icon, { [styles.active]: userVote === 'downvote' })}
          />
          {downvotes}
        </button>
      </footer>
    </Card>
  );
}
