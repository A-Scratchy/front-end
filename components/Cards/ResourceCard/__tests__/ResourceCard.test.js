/* eslint-env jest */
import React from 'react';
import createSnapshotTest from 'test-utils/createSnapshotTest';

import ResourceCard from '../ResourceCard';

describe('ResourceCard', () => {
  it('should render with required props', () => {
    createSnapshotTest(
      <ResourceCard
        imageSource="/static/images/icons/javascript_logo.svg"
        name="JavaScript for Dummies"
      />,
    );
  });
});
