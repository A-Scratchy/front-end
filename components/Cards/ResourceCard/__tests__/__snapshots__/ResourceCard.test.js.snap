// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResourceCard should render with required props 1`] = `
<article
  className="Card ResourceCard"
>
  <header
    className="titleSection"
  >
    <img
      alt="logo"
      src="/static/images/icons/javascript_logo.svg"
    />
    <h5>
      JavaScript for Dummies
    </h5>
  </header>
  <section>
    <p
      className="descriptionText"
    >
      
    </p>
  </section>
  <footer
    className="footerSection"
  >
    <span
      className="footerText"
    >
      I found this useful 
    </span>
    <button
      className="voteBtn faded"
      onClick={[Function]}
      type="button"
    >
      <svg
        className="icon"
      />
      0
    </button>
    <button
      className="voteBtn faded"
      onClick={[Function]}
      type="button"
    >
      <svg
        className="icon"
      />
      0
    </button>
  </footer>
</article>
`;
